sync/snap_server.go

```diff
@@ -149,72 +118,45 @@ func (b *snapServer) GetClassRange(ctx context.Context, request *spec.ClassRange
 			limitAddr = nil
 		}
 
-		// TODO: loop this
-		proofs, _, err := iterateWithLimit(ctrie, startAddr, limitAddr, determineMaxNodes(uint64(request.ChunksPerProof)), func(key, value *felt.Felt) error {
-			classkeys = append(classkeys, key)
-			return nil
-		})
-
-		coreclasses, err := b.blockchain.GetClasses(classkeys)
-		if err != nil {
-			yield(nil, err)
-			return
-		}
+		for {
+			proofs, finished, err := iterateWithLimit(ctrie, startAddr, limitAddr, determineMaxNodes(uint64(request.ChunksPerProof)), func(key, value *felt.Felt) error {
```

Comment:
`startAddr` need to change on each iteration so that it send out different portion of the range. Usually, it is set to the last leaf of the last iteration.

Reply:
I fixed `startAddr` assignment to the last visited node